doc/remediation-templating.md

```diff
@@ -93,305 +40,204 @@ spec:
-		return "Value-Not-Set"
-	}
-
-}
-
-func main() {
-	MachineConfigCM := "data:,...%0APrintMotd%20no%0A%0APrintLastLog%20yes%0A%23TCPKeepAlive%20yes%0APermitUserEnvironment%20no%0ACompression%20no%0AClientAliveInterval%20{{\"the_value_you_want_to_define65\" | findXCCDFValue | urlquery}}%0AClientAliveCountMax%200%0A%23UseDNS%20no%0A%23PidFile%20/var/run/sshd.pid%0A%23MaxStartups%2010%3A30%3A100%0A%23PermitTunnel%20no%0A%23ChrootDirectory%20none%0A%23..."
-	t, err := template.New("text").Funcs(template.FuncMap{"findXCCDFValue": findXCCDFValue}).
-		Parse(MachineConfigCM)
-	if err != nil {
-		panic(err)
-	}
-	t.Execute(os.Stdout, "nothinghere")
-
-  if(len(xccdf_variable)>0){
-    labels_var_set[compv1alpha1.RemediationVariableUsedLabel] = missing_value
-    remediation[compv1alpha1.RemediationXCCDFVariableUsedAnnotation] //from here we will save all the set xccdf variables
-    rem.SetLabels(labels)
-  }
-
-  if(len(missing_value) > 0){
-    labels[compv1alpha1.RemediationHasUnsetVariableLabel] = missing_value
-    remediation[compv1alpha1.RemediationHasUnsetVariableAnnotation] //from here we will save the not-set variables in the label
-    rem.SetLabels(labels)
-  }
-}
+The `source` above contains the file content, and the xccdf variables are often inside those file content. For example, there is an rule that configures auditd max_log_file_action Upon Reaching Maximum Log Size. We want to set `max_log_file_action` to a xccdf variable in file content like the following:
+
+```yml
+...
+priority_boost = 4
+name_format = hostname
+##name = mydomain
+max_log_file_action = {{.var_auditd_max_log_file_action}}
+space_left = {{.var_auditd_space_left}}
+space_left_action = {{.var_auditd_space_left_action}}
+verify_email = yes
+...
 ```
 
+To gernerate a remdiation for the rule, we can created `kubernets/shared.yml` file under that rule's folder, and save the content of file above to a variable to use as `source`, then call `kubernetes_machine_config_file(path='/etc/audit/auditd.conf', file_permissions_mode='0640', source=content_of_file` there.
 
-In the case, where an administrator has to set a value for the remediation but did not set one.
-In the aggregator, before we create the Remediation, if value not set or no default value found,
-we should add a label `compliance.openshift.io/has-unset-variable`, and `compliance.openshift.io/has-unset-xccdf-values` 
-annotation with the missing value variables. We should also have a label `compliance.openshift.io/variable-used` and a annotation
-`compliance.openshift.io/xccdf-variable-used`, And in the remediation controller, we should check the label,
-if label `compliance.openshift.io/has-unset-xccdf-variable` not empty, we should change
-`ComplianceRemediation` status to `Needs-Review`, and prevent it from applying.
+The file content `source` needs to get URL-encoded in `MachineConfig`, and in order to process the variable substitutions in the compliance operator, it needs to get URL-encoded decoded.
```

Comment:
I think just saying URL-decoded is easier than URL-encoded-decoded.

Reply:
That's a good point!